### ISSUE TITLE
The input of discriminator are different between CycleGAN paper and code

### ISSUE THREAD

**Issue by lscelory** [junyanz/pytorch-CycleGAN-and-pix2pix#701]
I found that in the CycleGAN paper section 3.1 Adversarial Loss, the author use GAN loss without condition. But after I read the code, I found actually the author takes the concatenate of real and fake as input of net D. So my question are:
1. What is the different between pix2pix and CycleGAN about the way we input to net D?(I understand the 'conditioned' in pix2pix but not in CycleGAN.)
2. If I use pix2pix or CycleGAN framework for a image transfer task(like change the pose) which input and output image are in the same domain, should I continue to use the conditioned input to discriminator?

Thank you!

**Comment by junyanz** [comment]
0 and 1: the code is consistent with the CycleGAN paper. We don't use conditional D for CycleGAN (as we don't have pairs). We use conditional D for pix2pix. Please refer to the original pix2pix paper for why conditional D is used in pix2pix.
2. It depends on your training data. If you don't have pairs, you cannot use conditional D. If you have pairs, you can use it and it may help.

**Comment by lscelory** [comment]
Thank you so much for your reply. After receiving your answer, I check my CycleGAN code in my computer and here. It's seem like I rewrote the input of D in CycleGAN using for a paired transfer task a long time ago and I forgot it. Sorry to bother you.
Still about question 2, is that the same person in same suit but different pose can be thought as a pair and use conditioned D? (Follow you kind advice I read the original pix2pix paper but I'm not sure is it the same condition with pix2pix.)

**Comment by junyanz** [comment]
Yes, it can be considered as paired data. See recent work on pose transfer [1](https://papers.nips.cc/paper/6644-pose-guided-person-image-generation.pdf), [2](https://arxiv.org/abs/1804.07739).

**Comment by lscelory** [comment]
Got it! Thanks again!
